examples/benchmarking_comparison_models.py

```diff
@@ -0,0 +1,209 @@
+    train_iterator = iterate_dataset(
+        training_scenarios,
+        groups_per_step=4,
+        initial_step=await model.get_step(),
+    )
+
+    for batch, epoch, global_step, epoch_step in train_iterator:
+        groups = await art.gather_trajectory_groups(
+            art.TrajectoryGroup(
+                (rollout(model, scenario) for _ in range(6)),
+            )
+            for scenario in batch
+        )
+        await model.train(groups)
+
+        if global_step % 20 == 0:
+            # Every 20 steps let's benchmark our model under training so we can
+            # see how it's doing.
+            await benchmark_model(model)
+
+    # At the end of training, let's benchmark the model again to see where it
+    # ended up.
+    await benchmark_model(model)
+
+
+async def main():
+    # Here we define a list of prompted comparison models we want to assess
+    # performance for.
+    gpt_4_1 = art.Model(
+        name="gpt-4.1",
+        project=PROJECT_NAME,
+        inference_base_url="https://api.openai.com/v1/",
+        inference_api_key=os.getenv("OPENAI_API_KEY"),
+    )
+
+    # We'll create a second GPT-4.1 variant that uses CoT reasoning
+    gpt_4_1_thinking = gpt_4_1.model_copy(deep=True)
+    gpt_4_1_thinking.name = "gpt-4.1-thinking"
+    assert isinstance(gpt_4_1_thinking.config, MyConfig)
+    gpt_4_1_thinking.config.use_thinking = True
+
+    gemini_flash = gpt_4_1.model_copy(deep=True)
+    gemini_flash.name = "gemini-flash-2.5"
+    gemini_flash.inference_base_url = "https://openrouter.ai/api/v1"
+    gemini_flash.inference_api_key = os.getenv("OPENROUTER_API_KEY")
+    gemini_flash.inference_model_name = "gemini/gemini-flash-2.5-flash-preview"
+
+    gemini_flash_thinking = gemini_flash.model_copy(deep=True)
+    gemini_flash_thinking.name = "gemini-flash-2.5-thinking"
+    assert isinstance(gemini_flash_thinking.config, MyConfig)
+    gemini_flash_thinking.config.use_thinking = True
+
+    prompted_models: list[art.Model] = [
+        gpt_4_1,
+        gpt_4_1_thinking,
+        gemini_flash,
+        gemini_flash_thinking,
+    ]
+
+    # We also can define the models we want to train.
+    qwen = art.TrainableModel(
+        name="qwen-2.5-14b-instruct",
+        project=PROJECT_NAME,
+        base_model="Qwen/Qwen2.5-14B-Instruct",
+        config=MyConfig(),
+    )
+    qwen_thinking = qwen.model_copy(deep=True)
+    assert isinstance(qwen_thinking.config, MyConfig)
+    qwen_thinking.name = "qwen-2.5-14b-instruct-thinking"
+
+    train_models: list[art.TrainableModel] = [
+        qwen,
+        qwen_thinking,
+    ]
+
+    # We need to register all our models with the local API so they're
+    # available for training and logging.
+    api = art.LocalAPI()
+    for model in train_models + prompted_models:
+        await model.register(api)
+
+    # For prompted models, we can benchmark them right away.
+    for model in prompted_models:
+        await benchmark_model(model)
```

Comment:
What do you think of encouraging the user to benchmark all models simultaneously? Same question for training multiple models simultaneously using SkypilotAPI. The difficulty of debugging an error might be higher (because it's less immediately clear what model was at fault, and logs might get confused across models) but the total run time should significantly decrease.

Reply:
Yeah makes sense, I've updated the script to work that way now. Slightly worried about readability for folks not super familiar with async/await but hopefully most people are.